plugins/modules/add_devices_to_group.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2022, Chris Santiago (resolutecoder@gmail.com)
+# MIT License (see LICENSE or https://opensource.org/licenses/MIT)
+
+from __future__ import absolute_import, division, print_function
+import json
+from ansible_collections.maxamillion.fleetmanager.plugins.module_utils.fleetmanager import (
+    ConsoleDotRequest,
+)
+from ansible.module_utils.six.moves.urllib.parse import quote
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import AnsibleModule
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: add_devices_to_group
+short_description: Create and remove devices to and from a group
+description:
+  - This module will create and remove devices to and from a group
+version_added: "0.1.0"
+options:
+  name:
+    description:
+      - Group name for adding devices to
+    required: true
+    type: str
+  devices:
+    description:
+      - List of devices to be added to a group
+    required: true
+    type: list
+  state:
+    description:
+      - Should the devices exist or not
+    required  true
+    type: str
+    choices: ['present', 'absent']
+
+author: Chris Santiago @resolutecoder
+"""
+
+EXAMPLES = """
+- name: Add devices to a group
+  hosts: CRC
+  gather_facts: false
+  tasks:
+    - name:
+      maxamillion.fleetmanager.add_devices_to_group:
+        name: 'ansible-group-santiago'
+        devices:
+          - 21402
+          - 21365
+          - 31601
+        state: 'present'
+"""
+
+
+def main():
+
+    EDGE_API_GROUPS = '/api/edge/v1/device-groups'
+
+    argspec = dict(
+        name=dict(required=True, type='str'),
+        devices=dict(required=True, type='list'),
+        state=dict(required=True, type='str')
```

Comment:
I should have mentioned this before, the [argspec](https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec) allows for some other validation options to automatically be taken care of and we should use the `choices` option here:

```python
state=dict(required=True, type='str', choices=['present', 'absent'])
```

This will ensure the playbook author provides a valid `state`.

Reply:
Awesome good catch, its removed :smile: